### PR TITLE
Add registration with redirect

### PR DESCRIPTION
**Pull request by VR333**
# REVIEWERS

- [ ] @vlodkopits 
- [ ] @StepanKovalyshyn 

# DESCRIPTION
Add component for registration using axios and devise gem, create controller with inheritance from device controller. Add json response to that controller. Modified devise configs to make device receive json. Add new key 'name' to permitted devise parameters. Also modified routes to match new controller. In general code is now able to register users with redirection to profile page and show errors if registration failed, for example due to incorrect name or email, cause they have been already used by someone else.

# TEST COVERAGE

no test coverage

# EXPERT REVIEW

[ ] @rokor

### CODE REVIEW

File: app/assets/stylesheets/persons.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the persons controller here.
+// They will automatically be included in application.css.
```
Comment (rokor) [inline app/assets/stylesheets/persons.scss]: do we need this file?

File: app/assets/stylesheets/registrations.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the registrations controller here.
```
Comment (rokor) [inline app/assets/stylesheets/registrations.scss]: do we need this file?

File: app/assets/javascripts/components/SignupForm.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        <div>
+          <form className="new_user"
+                id="new_user"
+                onSubmit={this.handleSubmit}
+                acceptCharset="UTF-8"
+                method="post">
+            <fieldset>
+              <legend>Sign Up</legend>
+              <div className="box boxx">
+                <input className="email"
```
Comment (rokor) [inline app/assets/javascripts/components/SignupForm.jsx]: inputs should be generated dynamically.

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+class RegistrationsController < Devise::RegistrationsController
+  def create
+    build_resource(sign_up_params)
+
+    if resource.save
+      if resource.active_for_authentication?
+        set_flash_message :notice, :signed_up if is_navigational_format?
```
Comment (rokor) [inline app/controllers/registrations_controller.rb]: from 7-9 and and 11-13 almost the same code.

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,3 +1,4 @@
 # ApplicationHelper
 module ApplicationHelper
+
```
Comment (rokor) [inline app/helpers/application_helper.rb]: do we need empty helper with comment?

File: app/views/devise/confirmations/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
```
Comment (rokor) [inline app/views/devise/confirmations/new.html.erb]: Do we need all this devise view are we using them?

File: spec/helpers/persons_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+# Specs in this file have access to a helper object that includes
+# the PersonsHelper. For example:
+#
+# describe PersonsHelper do
```
Comment (rokor) [inline spec/helpers/persons_helper_spec.rb]: commented code and pending folder, is this valid?

File: spec/helpers/registrations_helper_spec.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+# Specs in this file have access to a helper object that includes
+# the RegistrationsHelper. For example:
```
Comment (rokor) [inline spec/helpers/registrations_helper_spec.rb]: same comment as above

File: spec/views/persons/profile.html.erb_spec.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe "persons/profile.html.erb", type: :view do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (rokor) [inline spec/views/persons/profile.html.erb_spec.rb]: same, pending spec where the test or why we need them?

File: spec/views/players/create.html.erb_spec.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe "players/create.html.erb", type: :view do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (rokor) [inline spec/views/players/create.html.erb_spec.rb]: same, pending spec where the test or why we need them?

**Review by rokor — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/persons.scss (hunk 1 above)
Comment (VR333) [reply app/assets/stylesheets/persons.scss]: no we don't.

File: app/assets/stylesheets/registrations.scss (hunk 2 above)
Comment (VR333) [reply app/assets/stylesheets/registrations.scss]: no we don't.

File: app/controllers/registrations_controller.rb (hunk 4 above)
Comment (VR333) [reply app/controllers/registrations_controller.rb]: It is.

File: app/helpers/application_helper.rb (hunk 5 above)
Comment (VR333) [reply app/helpers/application_helper.rb]: no we don't.

File: app/views/devise/confirmations/new.html.erb (hunk 6 above)
Comment (VR333) [reply app/views/devise/confirmations/new.html.erb]: no we don't.

File: spec/helpers/persons_helper_spec.rb (hunk 7 above)
Comment (VR333) [reply spec/helpers/persons_helper_spec.rb]: no, It isn't.

File: spec/helpers/registrations_helper_spec.rb (hunk 8 above)
Comment (VR333) [reply spec/helpers/registrations_helper_spec.rb]: files were changed

File: spec/views/persons/profile.html.erb_spec.rb (hunk 9 above)
Comment (VR333) [reply spec/views/persons/profile.html.erb_spec.rb]: We don't need them, re-testing of basic rails functionality is useless

File: spec/views/players/create.html.erb_spec.rb (hunk 10 above)
Comment (VR333) [reply spec/views/players/create.html.erb_spec.rb]: same answer)

**Pull request closed without merging**